lib/snapshot/reader.js

```diff
@@ -9,21 +10,44 @@ module.exports = {
   processFunctionsToInspect,
 };
 
-function getLatest() {
-  return getFunctionsToInspect();
+function getLatest(local) {
+  let rawSnapshot;
+  const projectId = 'change me!';
+  if (local) {
+    debug('loading local snapshot');
+    rawSnapshot = fromLocal();
+  } else {
+    debug('loading snapshot from upstream');
+    rawSnapshot = fromUpstream(projectId);
```

Comment:
Missing `await`?

Reply:
removing this code altogether, I want to keep the agent's startup synchronous for the moment, and if I allow an async flow here then our e2e test breaks because we'll no longer promise 100% monitoring after `require`ing the agent.
the next phase will be an async flow in the background refreshing the data, and maybe later we'll also just be okay with an async agent start.